base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
```

Comment:
- [ ] Kindly remove this `super` because it causes errors about wrong arguments. we use super in the `initialize` when the class you inherit from has the `initialize` method also known as the constructor